src/components/Cluster/NewCluster/ReleaseSelector/ReleaseRow.tsx

```diff
@@ -0,0 +1,138 @@
+import { ClassNames } from '@emotion/core';
+import styled from '@emotion/styled';
+import { relativeDate } from 'lib/helpers';
+import PropTypes from 'prop-types';
+import React, { FC, useMemo, useState } from 'react';
+import Button from 'react-bootstrap/lib/Button';
+import { useSelector } from 'react-redux';
+import { getProvider } from 'selectors/mainInfoSelectors';
+import RadioInput from 'UI/Inputs/RadioInput';
+import ReleaseComponentLabel from 'UI/ReleaseComponentLabel';
+
+interface IReleaseRow extends IRelease {
+  isSelected: boolean;
+  selectRelease(releaseVersion: string): void;
+}
+
+const TableButton = styled(Button)`
+  height: 24px;
+  line-height: 24px;
+  position: relative;
+  top: -2px;
+  margin-left: 5px;
+  padding: 0px 15px;
+  i {
+    margin-right: 4px;
+  }
+`;
+
+const ComponentsWrapper = styled.div`
+  margin-left: ${({ theme }) => theme.spacingPx * 9}px;
+`;
+
+const Tr = styled.tr<{ isSelected: boolean }>`
+  background-color: ${({ isSelected, theme }) =>
+    isSelected ? theme.colors.foreground : 'transparent'};
+`;
+
+const StyledRadioInput = styled(RadioInput)`
+  margin-bottom: 0;
+`;
+
+const ReleaseRow: FC<IReleaseRow> = ({
+  components,
+  isSelected,
+  selectRelease,
+  timestamp,
+  version,
+}) => {
+  const provider = useSelector(getProvider);
+  const [collapsed, setCollapsed] = useState(true);
+  const kubernetesVersion = useMemo(
+    () =>
+      components.find((component) => component.name === 'kubernetes')?.version,
+    [components]
+  );
+
+  return (
+    <>
+      <Tr isSelected={isSelected} onClick={() => selectRelease(version)}>
+        <td>
+          <ClassNames>
+            {({ css }) => (
+              <StyledRadioInput
+                id={`select-${version}`}
+                title={`Select release v${version}`}
+                checked={isSelected}
+                value={isSelected ? 'true' : 'false'}
+                name={`select-${version}`}
+                onChange={() => selectRelease(version)}
+                bulletProps={{ className: css({ margin: 0 }) }}
+              />
+            )}
+          </ClassNames>
+        </td>
+        <td>v{version}</td>
+        <td>{relativeDate(timestamp)}</td>
+        <td>{kubernetesVersion}</td>
+        <td>
+          <TableButton
+            data-testid={`show-components-${version}`}
+            onClick={(e) => {
+              e.stopPropagation();
+              e.preventDefault();
+              setCollapsed(!collapsed);
+            }}
+          >
+            <i className={`fa fa-${collapsed ? 'eye' : 'eye-with-line'}`} />
+            Show
+          </TableButton>
+        </td>
+        <td>
+          <TableButton
+            data-testid={`open-changelog-${version}`}
+            href={`https://github.com/giantswarm/releases/blob/master/${provider}/v${version}/release-notes.md`}
```

Comment:
Can we move the link generation to a separate function?

Reply:
Sure, I'll create a hook for this.